nifi-nar-bundles/nifi-email-bundle/nifi-email-processors/src/main/java/org/apache/nifi/processors/email/ConsumeEWS.java

```diff
@@ -376,11 +387,22 @@ protected void fillMessageQueueIfNecessary(ProcessContext context) throws Proces
         }
     }
 
-    protected Folder getFolder(ExchangeService service) {
+    protected Folder getFolder(ExchangeService service, ProcessContext context) {
         Folder folder;
+        String MailboxProperty = context.getProperty(MAILBOX).getValue();
         if(folderName.equals("INBOX")){
             try {
-                folder = Folder.bind(service, WellKnownFolderName.Inbox);
+                if (!StringUtils.isEmpty(MailboxProperty))
+                    {
+                        Mailbox mailbox = new Mailbox(MailboxProperty);
+                        FolderId InboxFolderId = new FolderId(WellKnownFolderName.Inbox, mailbox);
```

Comment:
But now MAILBOX only works if the folder is the INBOX right? This would stop you from accessing any folder in a different mailbox except the INBOX.